src/databricks/labs/ucx/framework/owners.py

```diff
@@ -169,8 +171,15 @@ def get_workspace_administrator(self) -> str:
 class Ownership(ABC, Generic[Record]):
     """Determine an owner for a given type of object."""
 
-    def __init__(self, administrator_locator: AdministratorLocator) -> None:
+    _factories: dict[type, Ownership] = {}
+
+    @classmethod
+    def for_record_type(cls, record_type: type) -> Ownership[Record]:
+        return cast(Ownership[Record], cls._factories[record_type])
+
+    def __init__(self, administrator_locator: AdministratorLocator, record_type: type) -> None:
```

Comment:
this control flow confuses `pylint` and it will fail to detect hidden future bugs.wouldn't it be more clear if we add `is_applicable` abstract method

```
def is_applicable(self, record: Any) -> bool:
  return isinstance(record, Table)
```

this will:

1. avoid changing every constructor at the cost of adding new method
2. allow for one ownership to work with more than one record class
3. allow for an explicit injectable facade:

```python
class AnyOwnership(Ownership[Any]):
  def __init__(self, ownerships: list[Ownership]): ...

  def owner_of(self, record: Any) -> str:
    for o for self._ownerships: if o.is_applicable(record): return o.owner_of(record)
    return self._administrator_locator....
```

Reply:
done